addons/point_of_sale/static/src/js/models.js

```diff
@@ -746,7 +748,11 @@ exports.PosModel = Backbone.Model.extend({
         var self = this;
         this.set('synch',{ state: 'connecting', pending: orders.length});
 
-        return self._save_to_server(orders, options).done(function (server_ids) {
+        // Need to deep copy orders to avoid deleting new orders (which are saved to "orders" in localStorage also)
+        // while previous orders are in progress of "_save_to_order" and removed them after "_save_to_order" successfull
+        var flush_orders = JSON.parse(JSON.stringify(orders));
```

Comment:
I agree with the fact that there's an issue here, but I'm not sure yet we want to create a copy of the orders. Technically it's a relatively expensive operation and in the end we don't really need an entire copy except for the ids. Conceptually I'm not sure I like the idea. An issue like this should be solved by either some mutex or some other mechanism (like eg. checking what orders exactly have been synced to the server instead of deleting them all). I'll think about it further in the next coming days.